connect/runtime/src/main/java/org/apache/kafka/connect/runtime/distributed/DistributedHerder.java

```diff
@@ -1144,6 +1144,21 @@ public void putConnectorConfig(final String connName, final Map<String, String>
         );
     }
 
+    @Override
+    public void patchConnectorConfig(String connName, Map<String, String> configPatch, Callback<Created<ConnectorInfo>> callback) {
+        log.trace("Submitting connector config patch request {}", connName);
+        addRequest(() -> {
+            ConnectorInfo connectorInfo = connectorInfo(connName);
+            if (connectorInfo == null) {
+                callback.onCompletion(new NotFoundException("Connector " + connName + " not found", null), null);
+            } else {
+                Map<String, String> patchedConfig = applyConnectorConfigPatch(connectorInfo.config(), configPatch);
+                putConnectorConfig(connName, patchedConfig, true, callback);
```

Comment:
This adds another request to the queue instead of executing the follow-up logic immediately, which increases the chance for race conditions. Can we fix that? We could possibly have a synchronous variant of `putConnectorConfig` whose method body is pulled from [here](https://github.com/apache/kafka/blob/e63efbc5d771a6e69eb8678db6f88730b4e751e1/connect/runtime/src/main/java/org/apache/kafka/connect/runtime/distributed/DistributedHerder.java#L1099-L1140), and invoke that variant instead of the asynchronous one at this line in `patchConnectorConfig`.

Reply:
Yes, makes sense, done.